### PR TITLE
#26 add missing props

### PR DESCRIPTION
**Pull request by CAMOBAP**
- #26

### CODE REVIEW

File: Hcaptcha.d.ts [hunk 1]
```diff
@@ -41,9 +49,38 @@ type HcaptchaProps = {
...
-   * The webview style
+   * Enable / Disable sentry error reporting.
    */
-  style?: StyleProp<ViewStyle>
+  sentry?: boolean;
+  /**
+   * The url of api.js
+   * Default: https://js.hcaptcha.com/1/api.js (Override only if using first-party hosting feature.)
+   */
+  apiEndpoint?: string;
```
Comment (e271828-) [inline Hcaptcha.d.ts:59]:
```suggestion
  jsSdkPath?: string;
```

File: index.js [hunk 2]
```diff
@@ -93,15 +111,30 @@ ConfirmHcaptcha.propTypes = {
...
 ConfirmHcaptcha.defaultProps = {
+  size: 'invisible',
   passiveSiteKey: false,
   showLoading: false,
   backgroundColor: 'rgba(0, 0, 0, 0.3)',
   loadingIndicatorColor: null,
   theme: 'light',
   rqdata: null,
+  sentry: false,
+  apiEndpoint: 'https://hcaptcha.com/1/api.js',
```
Comment (e271828-) [inline index.js:132]:
```suggestion
  jsSdkPath: 'https://js.hcaptcha.com/1/api.js',
```

**Review by e271828- — COMMENTED** [review]
Let's rename apiEndpoint to something more accurate, like jsSdkPath

**Comment by CAMOBAP** [comment]
> Let's rename apiEndpoint to something more accurate, like jsSdkPath

I took the original `apiEndpoint` from android SDK, but agree `jsSdkPath` is more clear.

File: README.md [hunk 3]
```diff
@@ -108,18 +108,27 @@ Otherwise, you should pass in the preferred device locale, e.g. fetched from `ge
...
 | onMessage | Function (see [here](https://github.com/react-native-webview/react-native-webview/blob/master/src/WebViewTypes.ts#L299)) | The callback function that runs after receiving a response, error, or when user cancels. |
 | languageCode | string | Default language for hCaptcha; overrides phone defaults. A complete list of supported languages and their codes can be found [here](https://docs.hcaptcha.com/languages/) |
 | showLoading | boolean | Whether to show a loading indicator while the hCaptcha web content loads |
 | loadingIndicatorColor | string | Color of the ActivityIndicator |
 | backgroundColor | string | The background color code that will be applied to the main HTML element |
 | theme | string\|object | The theme can be 'light', 'dark', 'contrast' or a custom theme object (see Enterprise docs) |
 | rqdata | string | Hcaptcha execution options (see Enterprise docs) |
+| sentry | boolean | sentry error reporting (see Enterprise docs) |
+| apiEndpoint | string | The url of api.js. Default: https://js.hcaptcha.com/1/api.js (Override only if using first-party hosting feature.) |
```
Comment (CAMOBAP) [inline README.md:120]:
```suggestion
| jsSdkPath | string | The url of api.js. Default: https://js.hcaptcha.com/1/api.js (Override only if using first-party hosting feature.) |
```

File: Hcaptcha.js [hunk 4]
```diff
@@ -56,8 +72,15 @@ const Hcaptcha = ({
   backgroundColor,
   theme,
   rqdata,
+  sentry,
+  apiEndpoint,
```
Comment (CAMOBAP) [inline Hcaptcha.js:76]:
```suggestion
  jsSdkPath,
```

File: Hcaptcha.js [hunk 5]
```diff
@@ -56,8 +72,15 @@ const Hcaptcha = ({
...
+  endpoint,
+  reportapi,
+  assethost,
+  imghost,
+  host,
 }) => {
-  const apiUrl = buildHcaptchaApiUrl(siteKey, languageCode, theme);
+  const apiUrl = buildHcaptchaApiUrl(apiEndpoint, siteKey, languageCode, theme, host, sentry, endpoint, assethost, imghost, reportapi);
```
Comment (CAMOBAP) [inline Hcaptcha.js:83]:
```suggestion
  const apiUrl = buildHcaptchaApiUrl(jsSdkPath, siteKey, languageCode, theme, host, sentry, endpoint, assethost, imghost, reportapi);
```

File: Hcaptcha.js [hunk 6]
```diff
@@ -20,33 +20,49 @@ const patchPostMessageJsCode = `(${String(function () {
...
  * @param {*} languageCode: can be found at https://docs.hcaptcha.com/languages
  * @param {*} showLoading: loading indicator for webview till hCaptcha web content loads
  * @param {*} loadingIndicatorColor: color for the ActivityIndicator
  * @param {*} backgroundColor: backgroundColor which can be injected into HTML to alter css backdrop colour
- * @param {*} theme: can be 'light', 'dark', 'contrast' or custom theme object
- * @param {*} rqdata: see Enterprise docs
+ * @param {string|object} theme: can be 'light', 'dark', 'contrast' or custom theme object
+ * @param {string} rqdata: see Enterprise docs
+ * @param {boolean} sentry: sentry error reporting
+ * @param {string} apiEndpoint: The url of api.js. Default: https://js.hcaptcha.com/1/api.js (Override only if using first-party hosting feature.)
```
Comment (CAMOBAP) [inline Hcaptcha.js:56]:
```suggestion
 * @param {string} jsSdkPath: The url of api.js. Default: https://js.hcaptcha.com/1/api.js (Override only if using first-party hosting feature.)
```

File: Hcaptcha.js [hunk 7]
```diff
@@ -20,33 +20,49 @@ const patchPostMessageJsCode = `(${String(function () {
   window.ReactNativeWebView.postMessage = patchedPostMessage;
 })})();`;
 
-const buildHcaptchaApiUrl = (siteKey, languageCode, theme) => {
-  var url = 'https://hcaptcha.com/1/api.js?render=explicit&onload=onloadCallback';
-  url += `&host=${siteKey || 'missing-sitekey'}.react-native.hcaptcha.com`;
-  if (languageCode) {
-    url += '&hl=' + languageCode;
+const buildHcaptchaApiUrl = (apiEndpoint, siteKey, hl, theme, host, sentry, endpoint, assethost, imghost, reportapi) => {
+  var url = `${apiEndpoint || "https://hcaptcha.com/1/api.js"}?render=explicit&onload=onloadCallback`;
```
Comment (CAMOBAP) [inline Hcaptcha.js:24]:
```suggestion
const buildHcaptchaApiUrl = (jsSdkPath, siteKey, hl, theme, host, sentry, endpoint, assethost, imghost, reportapi) => {
  var url = `${jsSdkPath || "https://hcaptcha.com/1/api.js"}?render=explicit&onload=onloadCallback`;
```

**Comment by e271828-** [comment]
Ah, in that case we may want to opt for consistency, even though the naming isn't great.

**Comment by CAMOBAP** [comment]
> It looks like iOS SDK uses `jsSrc` which is much better, let's go with that.

Agree, @e271828- also I propose to rename it in android-sdk and make apiEndpoint deprecated, make sense?

File: .github/workflows/tests.yaml [hunk 8]
```diff
@@ -17,8 +17,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
-    - run: yarn
-    - run: yarn test
+    - run: npm install
+    - run: npm test
```
Comment (e271828-) [inline .github/workflows/tests.yaml:21]: Why did this switch to npm?

**Review by e271828- — APPROVED** [review]

**Comment by e271828-** [comment]
> > It looks like iOS SDK uses `jsSrc` which is much better, let's go with that.
> 
> Agree, @e271828- also I propose to rename it in android-sdk and make apiEndpoint deprecated, make sense?

Sounds good.

**Pull request merged**